packages/firestore/src/remote/serializer.ts

```diff
@@ -278,13 +265,35 @@ export class JsonProtoSerializer {
    * This method cheats. It's typed as returning "string" because that's what
    * our generated proto interfaces say bytes must be. But it should return
    * an Uint8Array in Node.
+   *
+   * Visible for testing.
    */
-  private toBytes(bytes: Blob): string {
+  toBytes(bytes: Blob | ByteString): string {
     if (this.options.useProto3Json) {
       return bytes.toBase64();
     } else {
       // The typings say it's a string, but it needs to be a Uint8Array in Node.
-      return this.unsafeCastProtoByteString(bytes.toUint8Array());
+      // Cast as string to avoid type check failing.
+      return (bytes.toUint8Array() as unknown) as string;
+    }
+  }
+
+  /**
+   * Returns a ByteString based on the proto string value.
+   * DO NOT USE THIS FOR ANYTHING ELSE.
+   * This method cheats. Value is typed as "string" because that's what
+   * our generated proto interfaces say bytes must be, but it is actually
+   * an Uint8Array in Node.
+   */
+  fromBytes(value: string | undefined): ByteString {
```

Comment:
This method could be properly typed and doesn't need to correlate to the Proto types (since `string` can be passed to `string | UInt8Array`). So this could just be:

```
  /**
   * Returns a ByteString based on the proto string value.
   */
  fromBytes(value: string | Uint8Array | undefined): ByteString {
    if (this.options.useProto3Json) {
      assert(typeof  value === 'string', "...");
      return ByteString.fromBase64String(value ? value : '');
    } else {
      assert(value === undefined || value instanceof Uint8Array, "...");
      return ByteString.fromUint8Array(
        value ? value: new Uint8Array()
      );
    }
  }
```